mock-core-configs/etc/mock/templates/centos-7.tpl

```diff
@@ -91,7 +97,8 @@ skip_if_unavailable=False
 {% if target_arch in ['x86_64', 'ppc64le', 'aarch64'] %}
 [centos-sclo-rh]
 name=CentOS-$releasever - SCLo rh
-mirrorlist=http://mirrorlist.centos.org/?release=7&arch=$basearch&repo=sclo-rh
+mirrorlist=http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=sclo-rh&infra=$infra
```

Comment:
What the `infra=$infra` means?

Reply:
It is part of the actual CentOS repository configurations. Mock already includes it for [CentOS 8](https://github.com/rpm-software-management/mock/blob/0fb40f3a3dc03864a6a23cdc5d2df8eaa3cf96b0/mock-core-configs/etc/mock/templates/centos-8.tpl#L31) and [CentOS Stream 8](https://github.com/rpm-software-management/mock/blob/0fb40f3a3dc03864a6a23cdc5d2df8eaa3cf96b0/mock-core-configs/etc/mock/templates/centos-stream-8.tpl#L43).

Reply:
Why do we need it here in mock?

Reply:
The repository configurations in CentOS are normally provided in the packages centos-release, epel-release, centos-release-scl, etc. But when the buildroot is first initialized, these packages aren't installed inside of it, so there is a bootstrapping problem. That's why mock has to write these repository configurations at all, so that it can create the buildroot.

As mentioned above, sometimes an administrator has to modify the provided configs in the OS, for example to point to a local mirror. The same changes would need to be made in the mock template, and this is far easier to do when it matches what is provided in CentOS.

So, to ask your question the other way around: why does mock need to use anything different than what is provided by CentOS?

Including an additional repository for koji is certainly understandable. But can we point to a reason that the URLs need to be different for the CentOS-provided repos?

Reply:
> The only reason distro repo configs are needed in mock at all is because they don't exist in the chroot when it is first created. Otherwise, the .repo files would already be provided by (in this case) the centos-release* packages.

Even if the centos-relese with repo files is provided we still intentionally use configs from mock.  This is
mostly an innocent bug (we waste the buildroot storage with useless repo files).

> To ask your question the other way around: why does mock need to use anything other than the URLs provided by CentOS for the CentOS repos? Are there specific reasons?

Any change needs to have a reason, as it forces developers to ask/focus to useless things.  So if we don't need the change, I'm -1.  I already asked several times about this, and haven't got the answer.